### PR TITLE
Add line info support.

### PR DESCRIPTION
**Pull request by zherczeg**
Add line info data to byte, which allows getting a backtrace info directly from the engine. Snapshots are not supported.

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-exceptions.c [hunk 1]
```diff
@@ -134,6 +138,23 @@ ecma_new_standard_error (ecma_standard_error_t error_type) /**< native error typ
 
   ((ecma_extended_object_t *) new_error_obj_p)->u.class_prop.class_id = LIT_MAGIC_STRING_ERROR_UL;
 
+#ifdef JERRY_ENABLE_LINE_INFO
+  /* Backtrace is not a magic string. */
+  const char *backtrace_str_p = "backtrace";
```
Comment (martijnthe) [inline jerry-core/ecma/operations/ecma-exceptions.c]: Would it make sense to name this `stack` instead so it's in line with what V8 and most JS engines do?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack

**Comment by zherczeg** [comment]
Snapshot issue: what happens if a snapshot with line info executed an engine without line info. Maybe we can think about it later, but this is a large enough patch.

**Comment by martijnthe** [comment]
> Snapshot issue: what happens if a snapshot with line info executed an engine without line info. 

Same answer as with RegExp literals: there needs to be a flag in the snapshot header to indicate that the feature is required.

> Maybe we can think about it later, but this is a large enough patch.

Sure, just wanted to understand whether there's an inherent limitation or something else (the latter it sounds like).

File: jerry-core/parser/js/js-parser-statm.c [hunk 2]
```diff
@@ -17,9 +17,9 @@
 
 #ifndef JERRY_DISABLE_JS_PARSER
 
-#ifdef JERRY_DEBUGGER
+#if defined JERRY_DEBUGGER || defined JERRY_ENABLE_LINE_INFO
```
[the remark this answers is not in the document]
Comment (zherczeg) [reply jerry-core/parser/js/js-parser-statm.c]: Is it a style rule? If it is, it is not consistent:
https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/ecma/base/ecma-literal-storage.c#L465

**Comment by yichoi** [comment]
LGTM

**Pull request merged**